src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConditionalRestrictionCheck.java

```diff
@@ -0,0 +1,178 @@
+
+    public static final String CONDITIONAL = ":conditional";
+    private static final List<String> RESTRICTION_TYPES = List.of("access", "restriction",
+            "maxspeed", "minspeed", "maxweight", "maxaxleload", "maxheight", "maxlength", "maxstay",
+            "maxgcweight", "maxgcweightrating", "interval", "duration", "overtaking", "oneway",
+            "fee", "toll", "noexit", "snowplowing", "disabled", "lanes", "parking");
+    private static final List<String> TRANSPORTATION_MODE = List.of("foot", "ski", "inline_skates",
+            "horse", "vehicle", "bicycle", "carriage", "trailer", "caravan", "motor_vehicle",
+            "motorcycle", "moped", "mofa", "motorcar", "motorhome", "tourist_bus", "coach", "goods",
+            "hgv", "hgv_articulated", "agricultural", "golf_cart", "atv", "snowmobile", "psv",
+            "bus", "minibus", "share_taxi", "taxi", "hov", "hazmat", "emergency", "canoe",
+            "electric_vehicle");
+    private static final List<String> DIRECTION = List.of("forward", "backward", "left", "right",
+            "both");
+    private static final List<String> ACCESS_RESTRICTION_VALUE = List.of("yes", "no", "private",
+            "permissive", "destination", "delivery", "customers", "designated", "use_sidepath",
+            "dismount", "agricultural", "forestry", "discouraged", "official", "lane");
+
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            "The conditional key {0} does not respect the \"<restriction-type>[:<transportation mode>][:<direction>]:conditional\" format",
+            "The conditional value {0} does not respect the format \"<restriction-value> @ <condition>[;<restriction-value> @ <condition>]\" ");
+    public static final int TWO_PARTS = 2;
+    public static final int THREE_PARTS = 3;
+    public static final int FOUR_PARTS = 4;
+
+    public ConditionalRestrictionCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // all elements that have a tag containing ":conditional"
+        return object.getOsmTags().keySet().stream().anyMatch(key -> key.contains(CONDITIONAL));
+    }
+
+    /**
+     * Checks if the conditional restrictions respects the format
+     *
+     * @param object
+     *            the atlas object containing a conditional tag
+     * @return an optional {@link CheckFlag} object
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final List<String> conditionalKeys = object.getOsmTags().keySet().stream()
+                .filter(key -> key.contains(CONDITIONAL)).collect(Collectors.toList());
+        for (final String key : conditionalKeys)
+        {
+            if (!isKeyValid(key))
+            {
+                return Optional.of(this.createFlag(object, this.getLocalizedInstruction(0, key)));
```

Comment:
By using a return statement here the check may miss flagging the value if it is also bad, and flagging other tags if an object has multiple bad conditional tags. Thus it could take multiple cycles of running the check and editing the feature to fully fix the problems. It would probably be better to add the instruction to a list of generated instructions for this feature, and flag the feature later if that list is not empty. You can see an example of this sort of arrangement in [RoundaboutValenceCheck](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutValenceCheck.java).